TrenDAP/Controllers/TrenDAPDBController.cs

```diff
@@ -133,14 +142,43 @@ public virtual ActionResult Get(int dataSourceID, string table = "")
                     return StatusCode(StatusCodes.Status500InternalServerError, ex);
                 }
             }
-
         }
 
-        private ActionResult GetOpenXDA(DataSource dataSource, string table) {
+        [HttpPost, Route("Channel/GetTrendChannels/{dataSourceID:int}")]
+        public virtual ActionResult Post(int dataSourceID, [FromBody] JObject filter)
+        {
+            using (AdoDataConnection connection = new AdoDataConnection(Configuration["SystemSettings:ConnectionString"], Configuration["SystemSettings:DataProviderString"]))
+            {
+                try
+                {
+                    DataSource dataSource = new TableOperations<DataSource>(connection).QueryRecordWhere("ID = {0}", dataSourceID);
+                    DataSourceHelper helper = new DataSourceHelper(dataSource);
+                    string type = connection.ExecuteScalar<string>("SELECT Name FROM DataSourceType WHERE ID = {0}", dataSource.DataSourceTypeID);
+                    Task<string> rsp;
+
+                    if (type == "TrenDAPDB")
+                    {
+                        rsp = helper.PostAsync("api/Channel/GetTrendSearchData", new StringContent(filter.ToString(), Encoding.UTF8, "application/json"));
+                    }
+                    else return StatusCode(StatusCodes.Status400BadRequest, "Datasource type not supported");
```

Comment:
Technically that does not mean the datasource is not supported but that this endpoint is not supposed to be used with any other datasourceType. I would return a 500 with a message to that effect